server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -284,6 +284,8 @@ public Iterator<Setting<?>> settings() {
     );
 
     public static final String SETTING_REMOTE_STORE_ENABLED = "index.remote_store.enabled";
+
+    public static final String SETTING_REMOTE_TRANSLOG_STORE_ENABLED = "index.remote.translog_store.enabled";
```

Comment:
Can it be `index.remote_store.translog.enabled`? It provides a nested structure to all the remote_store related settings.

```
{
  "settings": {
    "index": {
      "number_of_shards": 1,
      "number_of_replicas": 0,
      "remote_store": {
        "enabled": true,
        "repository": "repository-1",
        "translog": {
          "enabled": true
        }
      }
    }
  }
}
```

Reply:
Can we change remote segment store to `remote.segment_store `

Reply:
+1 - If this is tied to remote store `index.remote_store.translog.enabled` is more clear.